src/modules/tasksClass.js

```diff
@@ -0,0 +1,82 @@
+const newTaskDescription = document.querySelector('#new-task-title');
+const container = document.querySelector('.mylist');
+let checkboxes;
+let descriptions;
+
+export default class Tasks {
+  tasks;
+
+  constructor() {
+    this.getFromLocalStorage();
+  }
+
+  setLocalStorage = (newTasks) => localStorage.setItem('tasks', JSON.stringify(newTasks));
+
+  getFromLocalStorage = () => {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) ?? [];
+  };
+
+  updateTasks = () => {
+    let temp = this;
+    this.getFromLocalStorage();
+    container.innerHTML = '';
+    this.tasks.forEach((task, i) => {
+      task.index = i;
+      container.innerHTML += `<li>
+      <div class="task-container list-row">
+      <input class="checker" type="checkbox" id="${i}">
+      <input class="task-description" type="text" name="task-title" id="${i}" value="${task.description}"/>
+      <div class="icon-container delete task-icon-container" id="${i}"></div>
+      </div>
+    </li>`;
+    });
+    checkboxes = document.querySelectorAll('input[type=checkbox]');
+    checkboxes.forEach((checkbox, i) => {
+      if (temp.tasks[i].completed === true) {
+        checkbox.checked = true;
+        checkbox.nextSibling.nextSibling.classList.add('marked');
+      }
+      checkbox.addEventListener('change', () => {
+        if (checkbox.checked === true) {
+          checkbox.nextSibling.nextSibling.classList.add('marked');
+          temp.tasks[i].completed = true;
+        } else {
+          checkbox.nextSibling.nextSibling.classList.remove('marked');
+          temp.tasks[i].completed = false;
+        }
+        temp.setLocalStorage(temp.tasks);
+      });
+    });
+    descriptions = document.querySelectorAll('.task-description');
+    temp = this;
+    descriptions.forEach((desc, i) => {
+      desc.addEventListener('change', () => {
+        temp.tasks[i].description = desc.value;
+        temp.setLocalStorage(temp.tasks);
+      });
+    });
+  };
+
+  createTask = (e) => {
+    e.preventDefault();
+    const newTask = {
+      description: newTaskDescription.value,
+      completed: false,
+      index: this.tasks.length,
+    };
+    this.tasks.push(newTask);
+    this.clear();
+    this.setLocalStorage(this.tasks);
+    this.updateTasks();
+  };
```

Comment:
- [ ] When a task is added, its `index` property should be set to the value of the array length. So for example, when I add the first the `index` should `1`, the second task the `index` should be `2`, so on and so forth. Kindly fix that.

  ![add-](https://user-images.githubusercontent.com/36542986/184130534-40095a96-7d73-4a0e-a7f8-5bc424fac3d3.jpg)